src/lib/comms/sol-lwm2m.c

```diff
@@ -3778,25 +4248,59 @@ sol_lwm2m_client_start(struct sol_lwm2m_client *client)
         return -ENOENT;
     }
 
+    //Try to register with all available [non-bootstrap] servers
     SOL_VECTOR_FOREACH_IDX (&ctx->instances, instance, i) {
-        r = read_resources(client, ctx, instance, res, sol_util_array_size(res),
-            SECURITY_SERVER_URI, SECURITY_SERVER_IS_BOOTSTRAP,
-            SECURITY_SERVER_ID);
+        r = read_resources(client, ctx, instance, res, 1,
+            SECURITY_SERVER_IS_BOOTSTRAP);
         SOL_INT_CHECK_GOTO(r, < 0, err_exit);
 
         //Is it a bootstap?
-        if (!res[1].data[0].b) {
+        if (!res[0].data[0].b) {
+            sol_lwm2m_resource_clear(&res[0]);
+            r = read_resources(client, ctx, instance, res, 2,
+                SECURITY_SERVER_URI, SECURITY_SERVER_ID);
+            SOL_INT_CHECK_GOTO(r, < 0, err_exit);
+
             conn_ctx = server_connection_ctx_new(client, sol_str_slice_from_blob(res[0].data[0].blob),
-                res[2].data[0].integer);
+                res[1].data[0].integer);
             r = -ENOMEM;
-            SOL_NULL_CHECK_GOTO(conn_ctx, err_clear);
+            SOL_NULL_CHECK_GOTO(conn_ctx, err_clear_2);
             has_server = true;
+
+            clear_resource_array(res, 2);
+        } else {
+            sol_lwm2m_resource_clear(&res[0]);
+            bootstrap_account_idx = i;
```

Comment:
You shouldn't:

_"The LWM2M Client MUST have at most one LWM2M Bootstrap Server Account."_ [Spec, Pg. 19]

Basically, if you have more than one, you are already breaking the spec. The code will be nice and go on with the first one, though. :)